docs/modules/multi-tenancy/proc-multi-tenancy.adoc

```diff
@@ -0,0 +1,43 @@
+// file included in the following:
+//
+// assembly-multi-tenancy-filter.adoc
+
+[id='proc-multi-tenancy-{context}']
+= (Preview) Setting up the Multi-tenancy filter
+
+[role="_abstract"]
+This procedure describes how to set up the Multi-tenancy filter by configuring it in Kroxylicious.
+The filter dynamically prefixes resource names to create isolation between tenants using the same Kafka cluster.
+The prefix representing a tenant is taken from the name of the virtual cluster representing the tenant. 
+For example, if the virtual cluster is named `tenant-1`, the prefix is `tenant-1`.
+Each tenant must be represented by a unique virtual cluster that points to the same cluster.
+
+.Prerequisites
+
+* An instance of Kroxylicious. 
+For information on deploying Kroxylicious, see the link:{github}[samples and examples^].
+* A config map for Kroxylicious that includes the configuration for creating virtual clusters and filters. 
+* A virtual cluster definition for each tenant using the Kafka cluster. 
+You need at least two virtual clusters to apply multi-tenancy.
+
+.Procedure
+
+. Configure a `MultiTenantTransformationFilterFactory` type filter.
++
+[source, yaml]
+----
+filters:
+  - type: MultiTenantTransformationFilterFactory
+    config:
+      prefixResourceNameSeparator: "." # <1>
+----
+<1> The separator used for the prefix. 
+If a separator is not specified, `-` is the default.
++
+NOTE: Currently, only the prefix with separator is validated.
```

Comment:
What validation do we apply?  AFICT the only thing we validate is that the virtual cluster has a name.

I think we are missing some validation that the cluster name plus the separator produces a valid topic name for kafka.

Reply:
https://github.com/kroxylicious/kroxylicious/blob/3c17246e61585dd6f621887def6f7229b98574ae/kroxylicious-filters/kroxylicious-multitenant/src/main/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilter.java#L506 validates that the virtual cluster and separator together are valid as a topic name.   We don't currently validate if the resulting resource name falls within the length constraints.

Reply:
Yeah, your right I don't know why I didn't read it properly yesterday.